### PR TITLE
fix!: Migrate jobs away from deprecated interfaces

### PR DESCRIPTION
**Pull request by come-nc**
## Summary

Another nail in the coffin of `ILogger`.
See #32127 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/public/Log/ILogFactory.php [hunk 1]
```diff
@@ -40,15 +39,6 @@ interface ILogFactory {
 	 */
 	public function get(string $type): IWriter;
 
-	/**
-	 * @param string $path
-	 * @return ILogger
-	 * @since 14.0.0
```
Comment (ChristophWurst) [inline lib/public/Log/ILogFactory.php:46]: breaking. okay with me but unexpected for a chore commit

**Review by ChristophWurst — APPROVED** [review]

**Comment by ChristophWurst** [comment]
* Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required

This does need an entry in the upgrade docs.

File: lib/public/Log/ILogFactory.php (hunk 1 above)
Comment (come-nc) [reply lib/public/Log/ILogFactory.php:46]: `@deprecated 22.0.0 Use \OCP\Log\ILogFactory::getCustomPsrLogger`

I thought it was okay since it’s been deprecated since 22.
I’m a bit new to this chore/fix/enh thing, I tend to use chore when it’s not something user-facing that I’d want to list in the changelog, is there a better rationale?

Comment (ChristophWurst) [reply lib/public/Log/ILogFactory.php:46]: It's probably debatable if API changes should land in the changelog of server. Since it is a platform with a public API, I would say it should. The tools I've worked with tend to ignore *chore* for changelogs, even if the change is breaking (`chore!`). I'd say a `fix!` would be a good fit here.

https://www.conventionalcommits.org/en/v1.0.0/#summary

File: apps/dav/lib/BackgroundJob/RefreshWebcalJob.php [hunk 2]
```diff
@@ -34,50 +35,26 @@
...
-		$this->timeFactory = $timeFactory;
 	}
 
 	/**
 	 * this function is called at most every hour
 	 *
 	 * @inheritdoc
 	 */
-	public function execute(IJobList $jobList, ILogger $logger = null) {
+	public function start(IJobList $jobList): void {
```
Comment (nickvergessen) [inline apps/dav/lib/BackgroundJob/RefreshWebcalJob.php:57]: 🙈

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
above documentation PR looks like it's referencing the wrong thing.
This PR here should still be documented

**Comment by nickvergessen** [comment]
> Can you elaborate on this?

Let me rephrase then:

Even though `OC\BackgroundJob\…` were private, it was the (defacto) standard for quite some time.
I think it's worth to put a one-liner into https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_29.html#id4 for it.
